stytch/consumer/totps/types.go

```diff
@@ -0,0 +1,206 @@
+	// ignored. Total custom claims size cannot exceed four kilobytes.
+	SessionCustomClaims map[string]any `json:"session_custom_claims,omitempty"`
+}
+
+// CreateParams: Request type for `TOTPs.Create`.
+type CreateParams struct {
+	// UserID: The `user_id` of an active user the TOTP registration should be tied to.
+	UserID string `json:"user_id,omitempty"`
+	// ExpirationMinutes: The expiration for the TOTP instance. If the newly created TOTP is not authenticated
+	// within this time frame the TOTP will be unusable. Defaults to 60 (1 hour) with a minimum of 5 and a
+	// maximum of 1440.
+	ExpirationMinutes int32 `json:"expiration_minutes,omitempty"`
+}
+
+// RecoverParams: Request type for `TOTPs.Recover`.
+type RecoverParams struct {
+	// UserID: The `user_id` of an active user the TOTP registration should be tied to.
+	UserID string `json:"user_id,omitempty"`
+	// RecoveryCode: The recovery code to authenticate.
+	RecoveryCode string `json:"recovery_code,omitempty"`
+	// SessionToken: The `session_token` associated with a User's existing Session.
+	SessionToken string `json:"session_token,omitempty"`
+	// SessionDurationMinutes: Set the session lifetime to be this many minutes from now. This will start a new
+	// session if one doesn't already exist,
+	//   returning both an opaque `session_token` and `session_jwt` for this session. Remember that the
+	// `session_jwt` will have a fixed lifetime of
+	//   five minutes regardless of the underlying session duration, and will need to be refreshed over time.
+	//
+	//   This value must be a minimum of 5 and a maximum of 527040 minutes (366 days).
+	//
+	//   If a `session_token` or `session_jwt` is provided then a successful authentication will continue to
+	// extend the session this many minutes.
+	//
+	//   If the `session_duration_minutes` parameter is not specified, a Stytch session will not be created.
+	SessionDurationMinutes int32 `json:"session_duration_minutes,omitempty"`
+	// SessionJWT: The `session_jwt` associated with a User's existing Session.
+	SessionJWT string `json:"session_jwt,omitempty"`
+	// SessionCustomClaims: Add a custom claims map to the Session being authenticated. Claims are only created
+	// if a Session is initialized by providing a value in `session_duration_minutes`. Claims will be included
+	// on the Session object and in the JWT. To update a key in an existing Session, supply a new value. To
+	// delete a key, supply a null value.
+	//
+	//   Custom claims made with reserved claims ("iss", "sub", "aud", "exp", "nbf", "iat", "jti") will be
+	// ignored. Total custom claims size cannot exceed four kilobytes.
+	SessionCustomClaims map[string]any `json:"session_custom_claims,omitempty"`
+}
+
+// RecoveryCodesParams: Request type for `TOTPs.RecoveryCodes`.
+type RecoveryCodesParams struct {
+	// UserID: The `user_id` of an active user the TOTP registration should be tied to.
+	UserID string `json:"user_id,omitempty"`
+}
+
+// TOTP:
```

Comment:
Looks like the description for this one is below where we use it and there's nothing really to put here:
```
	// Totps: An array containing a list of all TOTP instances (along with their recovery codes) for a given
	// User in the Stytch API.
	Totps []TOTP `json:"totps,omitempty"`